syntax = "proto3";
option java_package = "com.bureauveritas.modelparser.model.proto";;

service RequestSerializer {
  rpc SerializeRequest (SerializeRequestRequest) returns (SerializeRequestResponse) {}
}

message SerializeRequestRequest {
  string model_str = 1;
  string operation_name = 2;
}

message SerializeRequestResponse {
  string url_path = 1;
  map<string,string> query_string = 2;
  string method = 3;
  map<string,string> headers = 4;
  bytes body = 5;
}
